includes/Checker/Checks/Trademarks_Check.php

```diff
@@ -0,0 +1,543 @@
+			$this->validate_slug_has_no_trademarks( $plugin_slug );
+		} catch ( Exception $e ) {
+			$this->add_result_error_for_file( $result, WP_PLUGIN_DIR . '/' . $result->plugin()->basename(), $e->getMessage() );
+		}
+	}
+
+	/**
+	 * Determines if we find a trademarked term in plugin name.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $plugin_name The plugin name.
+	 *
+	 * @throws Exception Thrown if we found trademarked term in plugin name.
+	 */
+	private function validate_name_has_no_trademarks( $plugin_name ) {
+		if ( empty( $plugin_name ) ) {
+			return;
+		}
+
+		$check = $this->has_trademarked_slug( $plugin_name );
+		if ( ! $check ) {
+			return;
+		}
+
+		if (
+			trim( $check, '-' ) === $check
+			&& in_array( $check, self::FOR_USE_EXCEPTIONS, true )
+		) {
+			// Trademarks that do NOT end in "-", but are within the FOR_USE_EXCEPTIONS array can be used, but only if it ends with 'for x'.
+			$message = sprintf(
+				/* translators: 1: plugin name, 2: found trademarked term */
+				__( 'The plugin name includes a restricted term. Your chosen plugin name - %1$s - contains the restricted term "%2$s" which cannot be used within in your plugin name, unless your plugin name ends with "for %2$s". The term must still not appear anywhere else in your name.', 'plugin-check' ),
+				'<code>' . esc_html( $plugin_name ) . '</code>',
+				esc_html( trim( $check, '-' ) )
+			);
+		} elseif ( trim( $check, '-' ) === $check ) {
+			// Trademarks that do NOT end in "-" indicate slug cannot contain term at all.
+			$message = sprintf(
+				/* translators: 1: plugin name, 2: found trademarked term */
+				__( 'The plugin name includes a restricted term. Your chosen plugin name - %1$s - contains the restricted term "%2$s" which cannot be used at all in your plugin name.', 'plugin-check' ),
+				'<code>' . esc_html( $plugin_name ) . '</code>',
+				esc_html( trim( $check, '-' ) )
+			);
+		} else {
+			// Trademarks ending in "-" indicate slug cannot BEGIN with that term.
+			$message = sprintf(
+				/* translators: 1: plugin name, 2: found trademarked term */
+				__( 'The plugin name includes a restricted term. Your chosen plugin name - %1$s - contains the restricted term "%2$s" and cannot be used to begin your plugin name. We disallow the use of certain terms in ways that are abused, or potentially infringe on and/or are misleading with regards to trademarks. You may use the term "%2$s" elsewhere in your plugin name, such as "... for %2$s".', 'plugin-check' ),
+				'<code>' . esc_html( $plugin_name ) . '</code>',
+				esc_html( trim( $check, '-' ) )
+			);
+		}
+
+		throw new Exception( $message );
+	}
+
+	/**
+	 * Determines if we find a trademarked term in plugin slug.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $plugin_slug The plugin slug.
+	 *
+	 * @throws Exception Thrown if we found trademarked term in plugin slug.
+	 */
+	private function validate_slug_has_no_trademarks( $plugin_slug ) {
+		if ( empty( $plugin_slug ) ) {
+			return;
+		}
```

Comment:
See above, this could be removed from here in favor of adding it in `has_trademarked_slug()`.